backend/src/test/java/gov/cdc/usds/simplereport/service/ApiUserServiceTest.java

```diff
@@ -100,6 +106,91 @@ void getUser_standardUser_error() {
+    initSampleData();
+
+    UserInfo newUserInfo =
+        _service.createUserInCurrentOrg(
+            "newuser@example.com",
+            new PersonName("First", "Middle", "Last", "Jr"),
+            Role.USER,
+            true);
+
+    assertEquals("newuser@example.com", newUserInfo.getEmail());
+
+    PersonName personName = newUserInfo.getNameInfo();
+    assertEquals("First", personName.getFirstName());
+    assertEquals("Middle", personName.getMiddleName());
+    assertEquals("Last", personName.getLastName());
+    assertEquals("Jr", personName.getSuffix());
+  }
+
+  @Test
+  @WithSimpleReportOrgAdminUser
+  void createUserInCurrentOrg_reprovisionDeletedUser_success() {
+    initSampleData();
+
+    // disable a user from this organization
+    ApiUser orgUser = _apiUserRepo.findByLoginEmail("nobody@example.com").get();
+    orgUser.setIsDeleted(true);
+    _apiUserRepo.save(orgUser);
+    _oktaRepo.setUserIsActive(orgUser.getLoginEmail(), false);
+
+    UserInfo reprovisionedUserInfo =
+        _service.createUserInCurrentOrg(
+            "nobody@example.com", new PersonName("First", "Middle", "Last", "Jr"), Role.USER, true);
+
+    // the user will be re-enabled and updated
+    assertEquals("nobody@example.com", reprovisionedUserInfo.getEmail());
+
+    PersonName personName = reprovisionedUserInfo.getNameInfo();
+    assertEquals("First", personName.getFirstName());
+    assertEquals("Middle", personName.getMiddleName());
+    assertEquals("Last", personName.getLastName());
+    assertEquals("Jr", personName.getSuffix());
+  }
+
+  @Test
+  @WithSimpleReportOrgAdminUser
+  void createUserInCurrentOrg_enabledUserExists_error() {
+    initSampleData();
+
+    PersonName personName = new PersonName("First", "Middle", "Last", "Jr");
+
+    ConflictingUserException caught =
+        assertThrows(
+            ConflictingUserException.class,
+            () ->
+                _service.createUserInCurrentOrg(
+                    "allfacilities@example.com", personName, Role.USER, true));
+
+    assertEquals("A user with this email address already exists.", caught.getMessage());
+  }
+
+  @Test
+  @WithSimpleReportOrgAdminUser
+  void createUserInCurrentOrg_disabledUserWrongOrg_error() {
+    initSampleData();
+
+    // disable a user from another organization
+    ApiUser wrongOrgUser = _apiUserRepo.findByLoginEmail("captain@pirate.com").get();
+    wrongOrgUser.setIsDeleted(true);
+    _apiUserRepo.save(wrongOrgUser);
+    _oktaRepo.setUserIsActive(wrongOrgUser.getLoginEmail(), false);
+
+    PersonName personName = new PersonName("First", "Middle", "Last", "Jr");
+
+    ConflictingUserException caught =
+        assertThrows(
+            ConflictingUserException.class,
+            () ->
+                _service.createUserInCurrentOrg("captain@pirate.com", personName, Role.USER, true));
+
+    assertEquals("The user already exists, but is not in your organization.", caught.getMessage());
```

Comment:
thinking more about this, should this kind of call result in an `AccessDeniedException`? that feels more appropriate.

Reply:
Good point, that does seem more appropriate!  You are also right that this error message would disclose more information that we should.

Reply:
Also, thinking about it more... Any error will probably give someone enough information to figure out what is happening if they care to investigate.  We don't need to tell them the reason explicitly, though.

`AccessDeniedException` elsewhere is used when a user tries to perform an action that they don't have authorization to do.  In this case, they do have permission for the action, just not with this specific data.  Since we have to fail anyway, how about failing with the same error message as when the user is in their org?  (as you originally suggested)

Reply:
"they do have permission for the action, just not with this specific data" is one of the standard use cases for `AccessDeniedException` -- e.g. trying to search on patients in a facility you don't belong to. i would prefer you follow that paradigm personally